### ISSUE TITLE
tarballs 404ing?

### ISSUE THREAD

**Issue by jonathankeebler** [npm/npm#12563]
A dependency of one of my packages, fibers v1.0.11, is 404ing on the tarball URL: https://registry.npmjs.org/fibers/-/fibers-1.0.11.tgz . However, when I look at https://registry.npmjs.org/fibers/ that is a valid tarbar URL. Is something going on with NPM or is it a problem with that package?

```
npm WARN package.json countries-cities@0.0.11 No repository field.
-npm ERR! fetch failed https://registry.npmjs.org/fibers/-/fibers-1.0.11.tgz
npm ERR! Error: 404 Not Found
npm ERR!     at WriteStream.<anonymous> (/home/rof/.nvm/v0.10.31/lib/node_modules/npm/lib/utils/fetch.js:58:12)
npm ERR!     at WriteStream.emit (events.js:117:20)
npm ERR!     at evalmachine.<anonymous>:1609:14
npm ERR!     at /home/rof/.nvm/v0.10.31/lib/node_modules/npm/node_modules/graceful-fs/graceful-fs.js:102:5
npm ERR!     at Object.oncomplete (evalmachine.<anonymous>:107:15)
npm ERR! If you need help, you may report this *entire* log,
npm ERR! including the npm and node versions, at:
npm ERR!     <http://github.com/npm/npm/issues>

npm ERR! System Linux 3.13.0-58-generic
npm ERR! command "/home/rof/.nvm/v0.10.31/bin/node" "/home/rof/.nvm/v0.10.31/bin/npm" "install"
npm ERR! cwd /home/rof/src/github.com/REDACTED
npm ERR! node -v v0.10.31
npm ERR! npm -v 1.4.23
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /home/rof/src/github.com/REDACTED/npm-debug.log
npm ERR! not ok code 0
```

**Comment by slavafomin** [comment]
I'm getting the exact same issue!

**Comment by slavafomin** [comment]
@laverdet any ideas?

**Comment by jonathankeebler** [comment]
@slavafomin I ended up removing the dep that needed that version, but I still think it's an NPM issue since the 404 references an internal path "https://www.npmjs.com/tmp/fibers-1.0.11.tgz_1462299523408_0.36823642742820084 does not exist"

**Comment by laverdet** [comment]
That's odd. I published a new version 15 minutes ago or so and then went to go get lunch. It passed all my local tests fine and npm publish exited cleanly so I didn't think to actually npm install it and test again.

I can republish in an hour or so, in the meantime just use 1.0.10?

**Comment by slavafomin** [comment]
Thanks @laverdet will be waiting forward for this!

However, how do I force `flightplan` to use `fibers `1.0.10` instead of `1.0.11`? I'm installing it globally with `npm i -g flightplan`.

**Comment by twhid** [comment]
@slavafomin if you install fibers@1.0.10 globally, I believe it should then satisfy the dep in flightplan but I haven't tested.

**Comment by devildeveloper** [comment]
Any official response ? ... i need that for my production server !!!

**Comment by hartaaron** [comment]
Is it the extra - in the url:

npm ERR! fetch failed https://registry.npmjs.org/fibers/-/fibers-1.0.11.tgz

**Comment by queso** [comment]
Seems fixed for me now when I try again.

**Comment by laverdet** [comment]
Adding my side of the story here for troubleshooting purposes.

npm guys--

I published fibers@1.0.11 earlier today using npm 3.8.6 & node 6.0.0. The publish command reported no errors and everything seemed to go smoothly. Here is the very boring output of that command:
```
marcel@marcel node-fibers $ npm publish
+ fibers@1.0.11
marcel@marcel node-fibers $
```

Shortly after I started receiving reports from users saying the new version failed to install. After running `npm cache clear` I was able to reproduce these errors. Somehow the package tarball didn't make it up online.

I then noticed a `npm-debug.log` had been created, that follows:
```
0 info it worked if it ends with ok
1 verbose cli [ '/opt/node-v6.0.0-x64/bin/node',
1 verbose cli   '/opt/local/bin/npm',
1 verbose cli   'publish' ]
2 info using npm@3.8.6
3 info using node@v6.0.0
4 verbose publish [ '.' ]
5 silly cache add args [ '.', null ]
6 verbose cache add spec .
7 silly cache add parsed spec Result {
7 silly cache add   raw: '.',
7 silly cache add   scope: null,
7 silly cache add   name: null,
7 silly cache add   rawSpec: '.',
7 silly cache add   spec: '/Users/marcel/bootcamp/node-fibers',
7 silly cache add   type: 'directory' }
8 verbose addLocalDirectory /Users/marcel/.npm/fibers/1.0.11/package.tgz not in flight; packing
9 verbose correctMkdir /Users/marcel/.npm correctMkdir not in flight; initializing
10 info lifecycle fibers@1.0.11~prepublish: fibers@1.0.11
11 silly lifecycle fibers@1.0.11~prepublish: no script for prepublish, continuing
12 verbose tar pack [ '/Users/marcel/.npm/fibers/1.0.11/package.tgz',
12 verbose tar pack   '/Users/marcel/bootcamp/node-fibers' ]
13 verbose tarball /Users/marcel/.npm/fibers/1.0.11/package.tgz
14 verbose folder /Users/marcel/bootcamp/node-fibers
15 verbose addLocalTarball adding from inside cache /Users/marcel/.npm/fibers/1.0.11/package.tgz
16 verbose correctMkdir /Users/marcel/.npm correctMkdir not in flight; initializing
17 silly cache afterAdd fibers@1.0.11
18 verbose afterAdd /Users/marcel/.npm/fibers/1.0.11/package/package.json not in flight; writing
19 verbose correctMkdir /Users/marcel/.npm correctMkdir not in flight; initializing
20 verbose afterAdd /Users/marcel/.npm/fibers/1.0.11/package/package.json written
21 silly publish { name: 'fibers',
21 silly publish   version: '1.0.11',
21 silly publish   description: 'Cooperative multi-tasking for Javascript',
21 silly publish   keywords:
21 silly publish    [ 'fiber',
21 silly publish      'fibers',
21 silly publish      'coroutine',
21 silly publish      'thread',
21 silly publish      'async',
21 silly publish      'parallel',
21 silly publish      'worker',
21 silly publish      'future',
21 silly publish      'promise' ],
21 silly publish   homepage: 'https://github.com/laverdet/node-fibers',
21 silly publish   author:
21 silly publish    { name: 'Marcel Laverdet',
21 silly publish      email: 'marcel@laverdet.com',
21 silly publish      url: 'https://github.com/laverdet/' },
21 silly publish   main: 'fibers',
21 silly publish   scripts:
21 silly publish    { install: 'node build.js || nodejs build.js',
21 silly publish      test: 'node test.js || nodejs test.js' },
21 silly publish   repository:
21 silly publish    { type: 'git',
21 silly publish      url: 'git://github.com/laverdet/node-fibers.git' },
21 silly publish   license: 'MIT',
21 silly publish   engines: { node: '>=0.5.2' },
```

My shell doesn't report non-0 error codes so I'm not sure if the `npm publish` command silently failed, but I assume it knew something was wrong because it dumped out this log file for me. Notably, the log file is missing "registryBase" operations, "sending authorization for write operation", and so on.

I bumped the version to 1.0.12 and ran `npm publish` and it worked the second time. No configuration changes had been made, besides I guess the aforementioned `npm cache clear`.

**Comment by othiym23** [comment]
@npm/registry-team has confirmed that the package tarball never hit npm's tarball servers as part of the publish process. Because this is something that happened on the registry, and isn't an issue with the CLI, I'm going to redirect you to the [`npm/registry` issue tracker](https://github.com/npm/registry/issues), where one of the members of that team will be able to provide you with further information about what happened here.

One a separate (but probably very related) note, @laverdet, see #5082 – due to a bug in npm's tarball creation code that is exacerbated by a performance improvement in Node.js 6.0, it's probably best to avoid using Node.js 6.x with npm while publishing (this is true irrespective of the npm CLI version). I'm working on the fix now, and am looking to include it in this week's release. My apologies for this mess, and I'm glad the subsequent publish of `fibers@1.0.12` was successful. Thanks to all for your time!
